modules/core/src/main/java/org/apache/synapse/config/xml/endpoints/EndpointDefinitionFactory.java

```diff
@@ -368,6 +444,14 @@ public EndpointDefinition createDefinition(OMElement elem) {
         return definition;
     }
 
+    private boolean isExpression(String expressionStr) {
+        Pattern pattern = Pattern.compile("\\{.*\\}");
+        if (pattern.matcher(expressionStr).matches()) {
```

Comment:
Can the expressionStr null? If so we need to handle it.

Reply:
Due to new changes this function is no longer used.